doc/migrations/4.0.1_4.x.x.md

```diff
@@ -6,3 +6,30 @@ Currently, only the `urn:xmpp:http:upload:0` XMLNS is served.
 All major, modern client libraries and applications support the 0.3.0+ specification.
 If you experience any issues with making requests to the HTTP File Upload service, please update your client.
 
+## Retirement of the old `*.cfg` format
+
+Since release 4.x.x, we are no longer supporting `*.cfg` MongooseIM configuration format. For further application's usage, please use `TOML` format.
+
+## Minor changes in the `TOML` config format
+
+* [`mod_bosh.max_pause`](../../modules/mod_bosh/#modulesmod_boshmax_pause) instead of `maxpause`
+
+* [`mod_disco.server_info.module`](../../modules/mod_disco/#modulesmod_discoserver_info): the field is optional, no longer required
+
+* [`mod_global_distrib.connections.advertised_endpoints`](../../modules/mod_global_distrib/#modulesmod_global_distribconnectionsadvertised_endpoints): default not set (`false` is no longer accepted)
+
+* `mod_global_distrib.connections.tls.enabled`: the flag was removed, TLS is enabled by providing the [`cacertfile`](../../modules/mod_global_distrib/#modulesmod_global_distribconnectionstlscacertfile) and [`certfile`](../../modules/mod_global_distrib/#modulesmod_global_distribconnectionstlscertfile) options
+
+* [`mod_http_upload.max_file_size`](../../modules/mod_http_upload/#modulesmod_http_uploadmax_file_size): `undefined` is no longer allowed
```

Comment:
Actually it seems that this was already changed (accidentally?) when introducing TOML, but I believe there were no comments about it, so let's keep it.